wordpress-mobile/WordPress-iOS
# CoreDataHelper: Extensions are way cooler

### Description:
In this PR we're converting `CoreDataHelper` into a `NSManagedObjectContext` extension, and implementing a mechanism that gives all NSManagedObject subclasses a default `class var entityName` (which can be overriden).

This was, initially, my goal. But Swift's compiler segmentation faults forced us to implement the helpers as a Struct.

This is Mark II, with a (hopefully) way more stable compiler.

### To test:
- Run the unit tests, verify that everything is green
- Smoke test the App (specially Notifications!)

Needs review: @astralbodies 
Thanks in advance sir!
